nixos/modules/services/misc/soft-serve.nix

```diff
@@ -0,0 +1,213 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.soft-serve;
+  configFile = format.generate "config.yaml" cfg.settings;
+  exe = getExe cfg.package;
+  format = pkgs.formats.yaml { };
+  user = "soft-serve";
+in
+{
+  options = {
+    services.soft-serve = {
+      enable = mkEnableOption "Enable soft-serve service";
+
+      package = mkPackageOption pkgs "soft-serve" { };
+
+      stateDir = mkOption {
+        type = types.path;
+        default = "/var/lib/soft-serve";
+        description = mdDoc ''
+          The absolute path to the data directory.
+
+          See <https://github.com/charmbracelet/soft-serve>.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = user;
+        description = mdDoc "User account under which soft-serve runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = user;
+        description = mdDoc "Group account under which soft-serve runs.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
+        description = mdDoc ''
+          The contents of the configuration file.
+
+          See <https://github.com/charmbracelet/soft-serve>.
+        '';
+        example = literalExpression ''
+          {
+            # Soft Serve Server configurations
+
+            # The name of the server.
+            # This is the name that will be displayed in the UI.
+            name = "Soft Serve";
+
+            # Log format to use. Valid values are "json", "logfmt", and "text".
+            log_format = "text";
+
+            # The SSH server configuration.
+            ssh = {
+              # The address on which the SSH server will listen.
+              listen_addr = ":23231";
+
+              # The public URL of the SSH server.
+              # This is the address that will be used to clone repositories.
+              public_url = "ssh://localhost:23231";
```

Comment:
Is that already in the range where any application can listen on?

Reply:
0 through 1023 are reserved.

Reply:
We can just use something in the range of 1024 to 9000 or so. The 5 digit ports are usually pretty wide open for anyone and are not really used by system services.